src/main/java/de/rondiplomatico/nds/polygon.java

```diff
@@ -0,0 +1,264 @@
+            { 6.0,  52.0},
+            { 6.1,  50.0},
+            { 8.0,  49.0},
+            { 7.5,  47.5},
+            {10.5,  45.9}
+        };
+        
+        // number of levels in map hierarchy
+        int maxLevels                               = 15;
+        // get a bounding octagonal envelope (defaults to quadrilateral in 2D case)
+        //         for sample data corresponding to a polygon (e.g. borders of a country)
+        NDSEnvelope envelope                        = new NDSEnvelope(polygonCoordinates);
+        // get corresponding bounding tile ID, i.e. find level where all polygon points are on the same tile
+        int[] masterTileInfo                        = envelope.getMasterTileInfo(maxLevels);
+        int masterTileLevel                         = masterTileInfo[0];
+        int masterTileID                            = masterTileInfo[1];
+        // store the master tile
+        NDSTile masterTile                          = new NDSTile(masterTileLevel, masterTileID);
+        // get all tiles covered by the polygon on the tstLevel
+        int tstLevel                                = 8; // 13
+        // refine polygon coordinates to avoid edges that are crossing a tile
+        // set refinement factor to -1 to adaptively refine
+        int numSamples                              = -1;
+        double[][] polygonCoordinatesRef            = polygon.refinePolygon(tstLevel, polygonCoordinates, numSamples);
+        // let's grab all tiles with polygon points
+        int[] uniqueTileIDs                         = polygon.getUniqueTileNumbersOnLevel(tstLevel, polygonCoordinatesRef);
+        // dump to image file for debugging
+        masterTile.printMap(tstLevel, uniqueTileIDs, "png", "map");
+        // get x/y tile indices for tiles with polygon points
+        // key is tileY, val is tileX (note: val may contain multiple tileX indices)
+        Map<Integer, List<Integer>> tileHM          = masterTile.tileNumbersToHM(tstLevel, uniqueTileIDs);
+        // dump to image file for debugging
+        Map<Integer, List<Integer>> filledTileHM    = polygon.mapFillPolygon(tileHM, uniqueTileIDs, "flood-fill-safe");
+        int[] filledTileIDs                         = masterTile.hmToTileNumbers(tstLevel, filledTileHM);
+        masterTile.printMap(tstLevel, filledTileIDs, "png", "mapFilled");
+        
+        long t1                                     = System.currentTimeMillis();
+        System.out.println("\n>>>INFO: Program finished in "+(t1-t0)+" ms.");
+    }
+    
+    /**
+     * Applies a flood-fill algorithm to a polygon
+     * 
+     * @param hmi
+     *              the hash map input hmi
+     * @param tileNumbers
+     *              the tileNumbers
+     * @param method
+     *              the flood-fill algorithm (flood-fill, row-fill)
+     * @return
+     */
+    public static Map<Integer, List<Integer>> mapFillPolygon(Map<Integer, List<Integer>> hmi, int[] tileNumbers, String method){
```

Comment:
for what do we have a "method" argument here? use two separate functions. (i've never seen that in my Java life :-))

also, as what can i interpret the results?

Reply:
I've opted for always including the safety net (`ff.fillHoles()`), because there's really no good way of deciding whether it's necessary or not. This makes the `method` modifier obsolete.